File: python/run-length-encoding/run_length_encoding.py
def decode(string):
    decoded = ''
    num_as_str = ''

    for i in range(len(string)):
        if string[i].isdigit():
            num_as_str = num_as_str + string[i]
        else:
            if num_as_str == '':
                num_as_str = '1'
            decoded = decoded + (string[i] * int(num_as_str))
            num_as_str = ''
        
    return decoded


def encode(string):
    encoded = ''
    num = 0

    for i in range(len(string)):
        if (i == (len(string) - 1)):
            if num == 0:
                encoded = encoded + string[i]
            else:
                encoded = encoded + str(num + 1) + string[i]
                num = 0
        elif string[i] != string[i+1]:
            if num == 0:
                encoded = encoded + string[i]
            else:
                encoded = encoded + str(num + 1) + string[i]
                num = 0
        else:
            num = num + 1

    return encoded

File: javascript/run-length-encoding/run-length-encoding.js
export const encode = input => {
  let encoded = "";
  let num = 0;
  let curr = "";

  [...input].map((char, index) => {
    if (index === input.length - 1 || char !== input[index + 1]) {
      num !== 0 ? (encoded += num + 1) : null;
      encoded += char;
      num = 0;
    } else {
      num += 1;
    }
  });

  return encoded;
};

export const decode = input => {
  let decoded = "";
  let num_as_str = "";

  [...input].map((char, index) => {
    if (char >= "0" && char <= "9") {
      num_as_str += char;
    } else {
      if (num_as_str === "") {
        num_as_str = "1";
      }
      decoded += char.repeat(num_as_str);
      num_as_str = "";
    }
  });

  return decoded;
};
